backend-project/small_eod/institutions/tests/test_serializers.py

```diff
@@ -0,0 +1,174 @@
+        self.default_serializer = AddressDataNestedSerializer(
+            data={
+                "email": "asd@asd.pl",
+                "city": "City1",
+                "epuap": "epuap",
+                "street": "street",
+                "house_no": "11",
+                "postal_code": "22222",
+                "flat_no": "",
+            }
+        )
+        self.obj = (
+            self.default_serializer.save()
+            if self.default_serializer.is_valid()
+            else None
+        )
+
+    def test_address_data_fields(self):
+        self.assertEqual(self.obj.email, "asd@asd.pl")
+        self.assertEqual(self.obj.city, "City1")
+        self.assertEqual(self.obj.epuap, "epuap")
+        data = AddressDataNestedSerializer(AddressData.objects.get()).data
+        self.assertEqual(data["street"], "street")
+        self.assertEqual(data["house_no"], "11")
+        self.assertEqual(data["postal_code"], "22222")
+        self.assertEqual(data["flat_no"], "")
+
+    def test_update(self):
+        serializer = AddressDataNestedSerializer(self.obj, data={"city": "New City"})
+        self.assertTrue(serializer.is_valid(), serializer.errors)
+        obj = serializer.save()
+        self.assertEqual(serializer.data["city"], "New City")
+        self.assertEqual(obj.city, "New City")
+
+
+class ExternalIdentifierSerializerTestCase(TestCase):
+    def setUp(self):
+        self.default_serializer = ExternalIdentifierNestedSerializer(
+            data={"nip": "1234567890", "regon": "1234567890"}
+        )
+        self.obj = (
+            self.default_serializer.save()
+            if self.default_serializer.is_valid()
+            else None
+        )
+
+    def test_external_identifier_fields(self):
+        self.assertEqual(self.obj.nip, "1234567890")
+        data = ExternalIdentifierNestedSerializer(ExternalIdentifier.objects.get()).data
+        self.assertTrue(data["regon"], "1234567890")
+
+    def test_field_validation_inheritance(self):
+        serializer = ExternalIdentifierNestedSerializer(
+            data={"nip": "12345678901", "regon": "1234567890"}
+        )
+        self.assertFalse(serializer.is_valid(), serializer.errors)
+
+    def test_update(self):
+        serializer = ExternalIdentifierNestedSerializer(
+            self.obj, data={"nip": "1111111111"}
+        )
+        self.assertTrue(serializer.is_valid(), serializer.errors)
+        obj = serializer.save()
+        self.assertEqual(serializer.data["nip"], "1111111111")
+        self.assertEqual(obj.nip, "1111111111")
+
+
+class InstitutionSerializerTestCase(TestCase):
+    def setUp(self):
+        self.user = UserFactory()
+        factory = APIRequestFactory()
+        self.request = factory.get("/")
+        force_authenticate(self.request, user=self.user)
+        self.request.user = self.user
+        self.category = CategoryFactory(level=3)
+        self.admin_unit = JednostkaAdministracyjnaFactory(category=self.category)
```

Comment:
```suggestion
        self.admin_unit = JednostkaAdministracyjnaFactory(category__level=self.category)
```

Generalnie w factory_boy jeżeli przekazujesz do niego dynamiczną wartość to warto się zastanowić czy poprawnie z niego korzystasz, czy nie powinieneś czegoś innego wygenerować, aby było krócej.

Reply:
Zrobione